sway-core/src/declaration_engine/declaration_engine.rs

```diff
@@ -41,6 +48,38 @@ impl DeclarationEngine {
         DeclarationId::new(self.slab.insert(declaration_wrapper), span)
     }
 
+    #[allow(clippy::map_entry)]
+    fn find_all_parents(&self, index: DeclarationId) -> Vec<DeclarationId> {
```

Comment:
Won't block this now, but some docstrings on these more dense algorithmic methods could help